skglm/solvers/cd_solver.py

```diff
@@ -355,7 +357,9 @@ def cd_solver(
                             print("Early exit")
                         break
         obj_out.append(p_obj)
-    return w, intercept, np.array(obj_out), stop_crit
+    if fit_intercept:
```

Comment:
since we need to extrapolate the intercept, what do you think of using a `w` of length `n_features + 1` inside the solver too ?

Reply:
so if fit_intercept=True, we would instantiate w of size n_features + 1 from the beginning? 
isn't it gonna be a problem for the warm_start in solver_path?